internal/test/test.go

```diff
@@ -92,7 +92,18 @@ func run(
 	}
 
 	if coverageReport != nil {
-		file, err := json.MarshalIndent(coverageReport, "", "  ")
+		var file []byte
+		var err error
+
+		ext := path.Ext(testFlags.CoverProfile)
+		if ext == ".json" {
+			file, err = json.MarshalIndent(coverageReport, "", "  ")
+		} else if ext == ".lcov" {
+			file, err = coverageReport.MarshalLCOV()
```

Comment:
Maybe instead of analyzing the extension, add a flag/option which allows the user to choose the format